### DISCUSSION TITLE
algoritmo de Luhn para comprobar la validez

### DISCUSSION THREAD

**Discussion by frankystanley** [community/community#63771] [Discussions]
### Select Topic Area

General

### Body

Quiero estas 2 soluciones si alguien me puede ayudar

Verificar si el número de tarjeta tiene exactamente 16 dígitos.
Aplicar el algoritmo de Luhn para comprobar la validez del número de tarjeta. El algoritmo funciona sumando los dígitos de posición par multiplicados por 2 (si el resultado es mayor que 9, se resta 9) y sumando los dígitos de posición impar. El resultado debe ser divisible por 10.

**Comment by EnriquezAlberto** [comment]
Fijate si con esto te puedo ayudar! Tuve un ejercicio parecido

```
function validarTarjetaCredito(numeroTarjeta) {
  const regexNumero = /^[0-9]{16}$/;
  if (!regexNumero.test(numeroTarjeta)) {
    return "Número de tarjeta inválido.";
  }

  const numeros = numeroTarjeta.split('').map(Number);
  const suma = numeros.reduce((total, num, index) => {
    if (index % 2 === 0) {
      num *= 2;
      if (num > 9) {
        num -= 9;
      }
    }
    return total + num;
  }, 0);

  if (suma % 10 !== 0) {
    return "Tarjeta no válida.";
  }

  return "Tarjeta válida.";
}
```

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
